### PR TITLE
Refactor generate logic, add disable logs, and disable from setup

### LINKED ISSUE #1554: Disable logs in generate for setup
Logs appear inside of the `generate` logic saying where they were created. There should be a way to disable logs in generate since the interactive setup uses its own logging.

### PR DESCRIPTION
**Pull request by chasefleming**
Closes #1554 

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: internal/super/generate.go [hunk 1]
```diff
@@ -102,7 +102,18 @@ func generateContract(
 	_ flowkit.Services,
 	state *flowkit.State,
 ) (result command.Result, err error) {
-	return generateNew(args, "contract", generateFlags.Directory, logger, state)
+	if len(args) < 1 {
```
Comment (ianthpun) [inline internal/super/generate.go:105]: shouldn't line 86 cover that already?

https://github.com/onflow/flow-cli/pull/1566/files#diff-1a34fb00f194ec131896919bfad4b3fdaf639c76fd91304228c346eb73dc89b9L86

Comment (chasefleming) [reply internal/super/generate.go]: Yes good catch! Removed

File: internal/super/generate.go [hunk 2]
```diff
@@ -122,64 +136,60 @@ func generateScript(
...
-	return strings.TrimSuffix(name, filepath.Ext(name))
+// TemplateMap defines a map of template types to their specific names
+type TemplateMap map[string]string
+
+type GeneratorOptions struct {
+	Directory   string
+	State       *flowkit.State
+	Logger      output.Logger
+	DisableLogs bool
 }
```
Comment (jribbink) [inline internal/super/generate.go:163]: Imo this is unnecessary & belongs in the `Generator` struct itself, and these can be set in the factory method.  Defining options like this isn't necessarily any safer/different than:

```
g := &Generator{
	Directory: xxx,
	State: xxx,
	Logger: xxx,
	DisableLogs: xxx,
}
```

And `GeneratorOptions` probably just adds an unnecessary layer of complexity.  The factory method can be used to enforce that a `&Generator` object is created with valid configuration.

e.g. `NewGenerator(directory, state, logger)`, `NewGeneratorWithoutLogs(directory, state)` -- although I _think_ that NOOP loggers is the idiomatic way to handle this.

File: internal/super/generate.go [hunk 3]
```diff
@@ -122,64 +136,60 @@ func generateScript(
...
+	return &Generator{
+		Options: options,
 	}
+}
 
-	var executedTemplate bytes.Buffer
-	// Execute the template with the provided data or nil if no data is needed
-	if err = tmpl.Execute(&executedTemplate, data); err != nil {
-		return "", fmt.Errorf("failed to execute template: %w", err)
+func (g *Generator) Create(typeNames TemplateMap) error {
```
Comment (jribbink) [inline internal/super/generate.go:175]: Is a map keyed by type the best data structure for this?  What happens if I want to generate two scripts?

File: internal/super/generate.go [hunk 4]
```diff
@@ -122,64 +136,60 @@ func generateScript(
...
-func stripCDCExtension(name string) string {
-	return strings.TrimSuffix(name, filepath.Ext(name))
+// TemplateMap defines a map of template types to their specific names
+type TemplateMap map[string]string
+
+type GeneratorOptions struct {
+	Directory   string
+	State       *flowkit.State
+	Logger      output.Logger
+	DisableLogs bool
```
Comment (jribbink) [inline internal/super/generate.go:162]: Could this be handled by a NOOP/nil logger?

File: internal/super/generate.go (hunk 3 above)
Comment (chasefleming) [reply internal/super/generate.go:160]: You could do it but we could use an array instead. Updated.

File: internal/super/generate.go (hunk 4 above)
Comment (chasefleming) [reply internal/super/generate.go:148]: Will that this in when we have support

File: internal/super/generate.go [hunk 5]
```diff
@@ -122,157 +133,185 @@ func generateScript(
...
-	}
-	return fmt.Sprintf("%s.cdc", name)
-}
+// TemplateMap defines a map of template types to their specific names
+type TemplateMap map[string][]string
 
-func stripCDCExtension(name string) string {
-	return strings.TrimSuffix(name, filepath.Ext(name))
+type Generator struct {
+	Directory   string
```
Comment (ianthpun) [inline internal/super/generate.go:145]: i believe we can make these private now since you got the constructor params workign :)

**Review by ianthpun — APPROVED** [review]

**Review by jribbink — APPROVED** [review]

**Pull request merged**